src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+#include <hyro/utils/SpinnerDefault.h>
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/SignalGenerator.h>
+#include <hyro/msgs/Signal.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+using namespace std::chrono_literals;
+
+namespace hyro
+{
+std::shared_ptr<HyroLogger> SignalGeneratorComponent::s_logger = HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+std::shared_ptr<ChannelOutput<Signal>> m_signal_output;
+hyro::DynamicPropertyAccess dynamic_property_access_signal("/signal"_uri);
+SignalGenerator sg;
+
+Result
+SignalGeneratorComponent::init (const ComponentConfiguration & configuration)
+{
```

Comment:
This component has four parameters **amplitude**, **frequency**, **name** and **cosine**.  These parameters must be initialized using the function ` config.parameters.getParameter`.

Reply:
Sorry, but I'm not exactly sure how to implement this. These variables are members of SignalGeneratorComponent class. Aren't they accessible when they're declared in the .h file?

Reply:
According to the exercise the **SignalGeneratorComponent** has four parameters (amplitude, frequency, name and cosine).

The parameters are define in the component connection string. Because of that, you need to call the function `config.parameter.getParameter` to load the parameters values.